core/templates/pages/exploration-editor-page/services/exploration-diff.service.ts

```diff
@@ -207,67 +207,70 @@ export class ExplorationDiffService {
       let changeList = changeListDatum.changeList;
       let directionForwards = changeListDatum.directionForwards;
       changeList.forEach(change => {
+        let explorationChangeAddState = change as ExplorationChangeAddState;
+
         if ((directionForwards && change.cmd === 'add_state') ||
             (!directionForwards && change.cmd === 'delete_state')) {
           if (!stateIds.hasOwnProperty(
-            (<ExplorationChangeAddState> change).state_name)) {
+            (explorationChangeAddState).state_name)
+          ) {
             let newId = this._generateNewId();
-            stateIds[(<ExplorationChangeAddState> change).state_name] = newId;
+            stateIds[(explorationChangeAddState).state_name] = newId;
```

Comment:
```suggestion
            stateIds[explorationChangeAddState.state_name] = newId;
```
ditto elsewhere in this file

Reply:
Not done.